sdk/aqueduct/artifacts/generic_artifact.py

```diff
@@ -7,34 +7,36 @@
 from aqueduct.artifacts.base_artifact import BaseArtifact
 from aqueduct.dag import DAG
 from aqueduct.enums import ArtifactType
-from aqueduct.error import AqueductError
+from aqueduct.error import AqueductError, InvalidArtifactTypeException
 
 
 class GenericArtifact(BaseArtifact):
     """This class represents a generic artifact within the flow's DAG.
-    Currently, a generic artifact can be any artifact other than table, numeric, bool, or parameter.
+    Currently, a generic artifact can be any artifact other than table, numeric, bool, or parameter
+    generated from eager execution, or an artifact of unknown type generated from lazy execution.
     """
 
     def __init__(
         self,
         dag: DAG,
         artifact_id: uuid.UUID,
```

Comment:
probably a separate thing - since it's already an `Artifact` object I'm curious if we should just use `id` and `type`

Reply:
I used to have `id` and `type` and vscode warned me that these are Python keywords...

Reply:
ahhh ouch, yeah this is quite brittle